controllers/clusterenvironment_controller.go

```diff
@@ -534,6 +546,52 @@ func (r *ClusterEnvironmentReconciler) reconcileApplicationNamespaces(ctx contex
 	return errors.Join(errcm, errct)
 }
 
+func (r *ClusterEnvironmentReconciler) checkExcessPermissions(ctx context.Context, cfg *rest.Config, ce *primazaiov1alpha1.ClusterEnvironment) error {
+	l := log.FromContext(ctx)
+	// check application namespaces permissions
+	apc := controlplane.NewAgentAppPermissionsChecker(cfg)
+	errs := []error{}
+
+	ep, err := apc.CheckExcessPermission(ctx, ce.Spec.ApplicationNamespaces)
+	if err != nil {
+		errs = append(errs, err)
+	} else if len(ep) > 0 {
+		m := metav1.Condition{
+			Type:    "ExcessPermissions",
+			Reason:  "ExcessPermissions",
+			Message: "More permissions than required granted for Application Namespaces",
+			Status:  "True",
+		}
+
+		meta.SetStatusCondition(&ce.Status.Conditions, m)
+		if err := r.Client.Status().Update(ctx, ce); err != nil {
+			l.Error(err, "error updating cluster environment status", "status", ce.Status)
+			errs = append(errs, err)
+		}
+	}
+
+	// check service namespaces permissions
+	spc := controlplane.NewAgentSvcPermissionsChecker(cfg)
+	ep, err = spc.CheckExcessPermission(ctx, ce.Spec.ServiceNamespaces)
+	if err != nil {
+		errs = append(errs, err)
+	} else if len(ep) > 0 {
```

Comment:
Just a refactor proposal

```suggestion
	errs := []error{}

	// check application namespaces permissions
	apc := controlplane.NewAgentAppPermissionsChecker(cfg)
	if ep, err := apc.CheckExcessPermission(ctx, ce.Spec.ApplicationNamespaces); err != nil {
		errs = append(errs, err)
	} else if len(ep) > 0 {
		m := metav1.Condition{
			Type:    "ExcessPermissions",
			Reason:  "ExcessPermissions",
			Message: "More permissions than required granted for Application Namespaces",
			Status:  "True",
		}

		meta.SetStatusCondition(&ce.Status.Conditions, m)
		if err := r.Client.Status().Update(ctx, ce); err != nil {
			l.Error(err, "error updating cluster environment status", "status", ce.Status)
			errs = append(errs, err)
		}
	}

	// check service namespaces permissions
	spc := controlplane.NewAgentSvcPermissionsChecker(cfg)
	if ep, err = spc.CheckExcessPermission(ctx, ce.Spec.ServiceNamespaces); err != nil {
		errs = append(errs, err)
	} else if len(ep) > 0 {
```